source/common/mqtt_connection.py

```diff
@@ -15,17 +17,40 @@ def _on_message(client, userdata, message):
     """
     try:
         payload = json.loads(str(message.payload.decode("utf-8")))
+        logging.info(f'Received message {payload} on topic {message.topic}')
 
         for fn in _topic_callbacks[message.topic]:
             fn(payload)        
     except:
         pass
 
-_client.on_message = _on_message
-_client.connect(_BROKER_ADDRESS)
-_client.loop_start()
+def _on_connect(client, userdata, flags, rc):
+    if rc == 0:
+        print("Connected to MQTT Broker!")
+    else:
+        print("Failed to connect, return code %d\n", rc)
 
 
+"""
+The paho mqtt clients must have different client ids. The broker disconnects the older client in favor of the new one
+if the clients ids are the same.
+https://stackoverflow.com/questions/40548730/two-paho-mqtt-clients-subscribing-to-the-same-client-localy
+As such, this function will load the client with the given cliend id, and each instance of a machine
+should have a different name. 
+Maybe we can achieve this by having a string of the form "client" + container ip
+For now, the client id is just the entity name and a random number appended
+"""
+def load_client(entity):
+    global _client
+
+    _client = mqtt.Client(entity + str(random.randint(a = int(1e5), b = int(1e10))))
```

Comment:
Aici cred ca ar fi bine sa nu punem doar un string random.
Cand primim un mesaj MQTT putem sa vedem cine l-a trimis, si deci si numele clientului.
Cred ca cel mai simplu ar fi ca atunci cand pornim un client din main.py sa mai necesite un parametru in plus, care sa fie id-ul acestuia.
Avantajul principal ar fi ca:
 * Nu mai apare fenomenul asta de randomness.
 * Putem sa identificam in mod unic fiecare client (de ex cand ii adaugam in baza de date).